test/replication-luatest/gh_7204_greeting_timeout_test.lua

```diff
@@ -0,0 +1,50 @@
+local fiber = require('fiber')
+local server = require('test.luatest_helpers.server')
+local socket = require('socket')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(g)
+    g.master_uri = server.build_instance_uri('master')
+    g.master = server:new({
+        alias = 'master',
+        box_cfg = {
+            replication_timeout = 0.1,
+        },
+    })
+    g.master:start()
```

Comment:
You probably don't need 2 instances. You can start one and right away try to set its replication to the URI where the fake server listens, then wait for the errors.

Reply:
You're right. I removed the master instance. Now the test looks much simpler:

<details><summary>Test</summary>

```Lua
local fiber = require('fiber')
local server = require('test.luatest_helpers.server')
local socket = require('socket')
local t = require('luatest')

local g = t.group()

g.before_all(function(g)
    g.server = server:new({
        alias = 'master',
        box_cfg = {
            replication_timeout = 0.1,
            replication_connect_timeout = 0.5,
        },
    })
    g.server:start()
end)

g.after_all(function(g)
    g.server:drop()
end)

g.test_greeting_timeout = function(g)
    local uri = server.build_instance_uri('server')
    local s = socket.tcp_server('unix/', uri, {
        handler = function() fiber.sleep(9000) end
    })
    t.assert(s)
    g.server:exec(function(uri)
        box.cfg{replication = uri}
    end, {uri})
    t.helpers.retrying({}, function()
        t.assert(g.server:grep_log('timed out'))
        t.assert(g.server:grep_log('will retry'))
    end)
    s:close()
end
```

</details>